src/libp2p_yamux_session.erl

```diff
@@ -359,6 +388,12 @@ ping_receive(#header{length=PingID}, State=#state{pings=Pings}) ->
     % with the ping time
     case maps:take(PingID, Pings) of
         error -> State;
+        {{liveness, TimerRef, _}, NewPings} ->
+            %% When we receive a ping sent from a liveness check we
+            %% jus tcancel the timer and kick the liveness timer down
+            %% the road again.
+            erlang:cancel_timer(TimerRef),
```

Comment:
isn't the cancel here redundant given the kick also cancels it?

Reply:
oh sorry, this is another timer.

Reply:
It is another timer indeed. Good eye